src/linter/ui5Types/BestPractices.ts

```diff
@@ -0,0 +1,345 @@
+			instanceOfPropsRecord(manifestSapui5Section?.routing?.value) &&
+			instanceOfPropsRecord(manifestSapui5Section?.routing?.value.config?.value) &&
+			typeof manifestSapui5Section?.routing?.value.config?.value.async?.value === "boolean") {
+			routingAsyncFlag = manifestSapui5Section?.routing?.value.config?.value.async?.value;
+		}
+	} else {
+		const parsedManifestContent =
+			JSON.parse(manifestContent ?? "{}") as SAPJSONSchemaForWebApplicationManifestFile;
+
+		const {rootView, routing} = parsedManifestContent["sap.ui5"] ?? {} as JSONSchemaForSAPUI5Namespace;
+		// @ts-expect-error async is part of RootViewDefFlexEnabled and RootViewDef
+		rootViewAsyncFlag = rootView ? rootView.async as boolean | undefined : rootViewAsyncFlag;
+		routingAsyncFlag = routing?.config ? routing.config.async : routingAsyncFlag;
+
+		hasManifestDefinition = !!(componentManifest &&
+		ts.isPropertyAssignment(componentManifest) &&
+		componentManifest.initializer.getText() === "\"json\"");
+	}
+
+	return {
+		routingAsyncFlag,
+		rootViewAsyncFlag,
+		hasAsyncInterface,
+		hasManifestDefinition,
+	};
+}
+
+function extractPropsRecursive(node: ts.ObjectLiteralExpression) {
+	const properties = Object.create(null) as propsRecord;
+
+	node.properties?.forEach((prop) => {
+		if (!ts.isPropertyAssignment(prop) || !prop.name) {
+			return;
+		}
+
+		const key = prop.name.getText();
+		if (prop.initializer.kind === ts.SyntaxKind.TrueKeyword) {
+			properties[key] = {value: true, node: prop.initializer};
+		} else if (prop.initializer.kind === ts.SyntaxKind.FalseKeyword) {
+			properties[key] = {value: false, node: prop.initializer};
+		} else if (prop.initializer.kind === ts.SyntaxKind.NullKeyword) {
+			properties[key] = {value: null, node: prop.initializer};
+		} else if (ts.isObjectLiteralExpression(prop.initializer) && prop.initializer.properties) {
+			properties[key] = {value: extractPropsRecursive(prop.initializer), node: prop.initializer};
+		} else if (ts.isArrayLiteralExpression(prop.initializer)) {
+			const resolvedValue = prop.initializer.elements.map((elem) => {
+				if (!ts.isObjectLiteralExpression(elem)) {
+					return;
+				}
+				return extractPropsRecursive(elem);
+			}).filter(($) => $) as propsRecordValueType[];
+
+			properties[key] = {value: resolvedValue, node: prop.initializer};
+		} else if (
+			(ts.isIdentifier(prop.initializer) ||
+			ts.isNumericLiteral(prop.initializer) ||
+			ts.isStringLiteral(prop.initializer)) &&
+
+			prop.initializer.getText()) {
+			properties[key] = {value: prop.initializer.getText(), node: prop.initializer};
+		} else {
+			// throw new Error("Unhandled property assignment");
+		}
+	});
+	return properties;
+}
+
+function reportResults(
+	analysisResult: AsyncInterfaceFindType,
+	reporter: SourceFileReporter,
+	classDesc: ts.ClassDeclaration
+) {
+	const {hasAsyncInterface, routingAsyncFlag, rootViewAsyncFlag, hasManifestDefinition} = analysisResult;
+
+	if (!hasManifestDefinition) {
+		reporter.addMessage({
```

Comment:
There's a false positive finding when the manifest is declared this way:
```js
metadata: {
  "manifest": "json"
}
```
and no finding (expected) when it's defined like this
```js
metadata: {
  manifest: "json"
}
```

Reply:
Thanks! Fixed with https://github.com/SAP/ui5-linter/pull/73/commits/af1d2f9d21cb8a083405848b3a5f6bf27f0866b4